### PR TITLE
Rewritten the Pillow-SIMD readme

### PR DESCRIPTION
**Pull request by dayton1987**
Fixes # .

Changes proposed in this pull request:
- 
- 
-

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,69 +1,71 @@
 # Pillow-SIMD
 
-Pillow-SIMD is "following" Pillow fork (which is PIL fork itself).
-"Following" means than Pillow-SIMD versions are 100% compatible
-drop-in replacement for Pillow with the same version number.
-For example, `Pillow-SIMD 3.2.0.post3` is drop-in replacement for
-`Pillow 3.2.0` and  `Pillow-SIMD 3.3.3.post0` for `Pillow 3.3.3`.
+Pillow-SIMD is "following" the Pillow fork (which is a PIL's fork itself).
```
Comment (homm) [inline README.md]: – fork
– the

File: README.md [hunk 2]
```diff
@@ -1,69 +1,71 @@
...
-by sequential box filters. But a number of such improvements are very limited.
-It is also very tempting to use more processor unit resources 
-(via parallelization) when they are available. But it is handier just
-to make things faster on the same resources. And that is where SIMD works better.
+There are multiple ways to tweak image processing performance.
+To name a few, such ways can be: utilizing better algorithms, optimizing existing implementations, 
+using more processing power and/or resources. 
+One of the great examples of using a more efficient algorithm is [replacing][gaussian-blur-changes] 
+a convolution-based Gaussian blur with a sequential-box one. 
+Such examples are rather rare, though. It is also known, that certain processes might be optimized
```
Comment (homm) [inline README.md:23]: cr

File: README.md [hunk 3]
```diff
@@ -1,69 +1,71 @@
...
-Modern CPU have different SIMD instructions sets like
-MMX, SSE-SSE4, AVX, AVX2, AVX512, NEON.
+SIMD stands for "single instruction, multiple data" and its essence is 
+in performing the same operation on multiple data points simultaneously 
+by using multiple processing elements. 
+Common CPU SIMD instruction sets are MMX, SSE-SSE4, AVX, AVX2, AVX512, NEON.
 
-Currently, Pillow-SIMD can be [compiled](#installation) with SSE4 (default)
-and AVX2 support.
+Currently, Pillow-SIMD can be [compiled](#installation) with SSE4 (default) and AVX2 support.
```
Comment (homm) [inline README.md]: or

File: README.md [hunk 4]
```diff
@@ -1,69 +1,71 @@
...
 
 
 ## Status
 
+Pillow-SIMD project is production-ready for you to start building SIMD-enabled image processing systems.
```
Comment (homm) [inline README.md]: –  for you to start building SIMD-enabled image processing systems

File: README.md [hunk 5]
```diff
@@ -1,69 +1,71 @@
...
+The project is supported by Uploadcare, a SAAS for cloud-based image storing and processing.
 [![Uploadcare][uploadcare.logo]][uploadcare.com]
+In fact, Uploadcare itself has been running Pillow-SIMD for about a year now.
```
Comment (homm) [inline README.md]: – itself

Comment (homm) [reply README.md]: 2 года

File: README.md [hunk 6]
```diff
@@ -1,69 +1,71 @@
...
 
-Pillow-SIMD can be used in production. Pillow-SIMD has been operating on
-[Uploadcare][uploadcare.com] servers for more than 1 year.
-Uploadcare is SAAS for image storing and processing in the cloud
-and the main sponsor of Pillow-SIMD project.
-
-Currently, following operations are accelerated:
+The following Uploadcare image operations are currently SIMD-accelerated:
```
Comment (homm) [inline README.md]: – Uploadcare

File: README.md [hunk 7]
```diff
@@ -1,69 +1,71 @@
...
 
 
 ## Benchmarks
 
-The numbers in the table represent processed megapixels of source RGB 2560x1600
-image per second. For example, if resize of 2560x1600 image is done
-in 0.5 seconds, the result will be 8.2 Mpx/s.
+In order for you to clearly assess the productivity of implementing SIMD computing into Pillow image processing, 
+we ran a number of benchmarks. The respective results can be found in the table below. 
+The numbers represent processing rates in megapixels per second (Mpx/s).
```
Comment (homm) [inline README.md:59]: (the more — the better)

File: README.md [hunk 8]
```diff
@@ -1,69 +1,71 @@
...
+The numbers represent processing rates in megapixels per second (Mpx/s). 
+For instance, the rate at which a 2560x1600 RGB image is processed in 0.5 seconds equals to 8.2 Mpx/s.
+Here are the instruments we've been up to during the benchmarks:
 
 - Skia 53
 - ImageMagick 6.9.3-8 Q8 x86_64
 - Pillow 3.4.1
 - Pillow-SIMD 3.4.1.post1
 
+Now, let's proceed to the numbers (the more — the better):
```
Comment (homm) [inline README.md]: – – –

File: README.md [hunk 9]
```diff
@@ -1,69 +1,71 @@
...
 ## Benchmarks
 
-The numbers in the table represent processed megapixels of source RGB 2560x1600
-image per second. For example, if resize of 2560x1600 image is done
-in 0.5 seconds, the result will be 8.2 Mpx/s.
+In order for you to clearly assess the productivity of implementing SIMD computing into Pillow image processing, 
+we ran a number of benchmarks. The respective results can be found in the table below. 
+The numbers represent processing rates in megapixels per second (Mpx/s). 
+For instance, the rate at which a 2560x1600 RGB image is processed in 0.5 seconds equals to 8.2 Mpx/s.
+Here are the instruments we've been up to during the benchmarks:
```
Comment (homm) [inline README.md]: the main idea is libs versions

File: README.md [hunk 10]
```diff
@@ -83,89 +85,87 @@ Operation               | Filter  | IM   | Pillow| SIMD SSE4| SIMD AVX2| Skia 53
...
 
 
-### Some conclusion
+### A brief conclusion
 
-Pillow is always faster than ImageMagick. And Pillow-SIMD is faster
-than Pillow in 4—5 times. In general, Pillow-SIMD with AVX2 always
-**16-40 times faster** than ImageMagick and overperforms Skia,
-high-speed graphics library used in Chromium, up to 2 times.
+The results show that Pillow is generally faster than ImageMagick,
```
Comment (homm) [inline README.md]: revert to always

File: README.md [hunk 11]
```diff
@@ -83,89 +85,87 @@ Operation               | Filter  | IM   | Pillow| SIMD SSE4| SIMD AVX2| Skia 53
...
-All tests were performed on Ubuntu 14.04 64-bit running on
-Intel Core i5 4258U with AVX2 CPU on the single thread.
-
-ImageMagick performance was measured with command-line tool `convert` with
-`-verbose` and `-bench` arguments. I use command line because
-I need to test the latest version and this is the easiest way to do that.
-
-All operations produce exactly the same results.
+All rates were measured using the following setup: Ubuntu 14.04 64-bit, 
+single-thread AVX2-enabled intel i5 4258U CPU.
```
Comment (homm) [inline README.md]: Intel

File: README.md [hunk 12]
```diff
@@ -83,89 +85,87 @@ Operation               | Filter  | IM   | Pillow| SIMD SSE4| SIMD AVX2| Skia 53
...
-for *gaussian blur*, because if the radius is too small, this is *not*
-gaussian blur anymore. And if the radius is big this does not give any
-advantages but makes operation slower. For the test, I set the radius
-to sigma × 2.5.
+In ImageMagick, Gaussian blur operation invokes two parameters: 
+the first is called 'radius' and the second is called 'sigma'.
+In fact, in order for the blur operation to be Gaussian, there should be no additional parameters. 
+When the radius value is too small the blur procedure ceases to be Gaussian and 
+if the value is excessively big the operation gets slowed down with zero benefits in exchange. 
+For the benchmarking purposes, the radius was set to sigma × 2.5.
```
Comment (homm) [inline README.md]: `sigma × 2.5`

File: README.md [hunk 13]
```diff
@@ -83,89 +85,87 @@ Operation               | Filter  | IM   | Pillow| SIMD SSE4| SIMD AVX2| Skia 53
...
 
-There are no cheats. High-quality resize and blur methods are used for all
-benchmarks. Results are almost pixel-perfect. The difference is only effective
-algorithms. Resampling in Pillow was rewritten in version 2.7 with 
-minimal usage of floating point numbers, precomputed coefficients and
-cache-awareness transposition. This result was improved in 3.3 & 3.4 with
+No cheats involved. We've used identical high-quality resize and blur methods for the benchmark. 
+Outcomes produced by different libraries are in almost pixel-perfect agreement. 
+The difference in measured rates is only provided with the performance of every involved algorithm. 
+Resampling for Pillow 2.7 was rewritten with minimal usage of floating point calculations,
```
Comment (homm) [inline README.md]: – – –

Comment (homm) [reply README.md]: and below

File: README.md [hunk 14]
```diff
@@ -83,89 +85,87 @@ Operation               | Filter  | IM   | Pillow| SIMD SSE4| SIMD AVX2| Skia 53
...
+The results were further improved in versions 3.3 & 3.4 by utilizing
 integer-only arithmetics and other optimizations.
 
-
 ## Why Pillow-SIMD is even faster
 
-Because of SIMD, of course. But this is not all. Heavy loops unrolling,
-specific instructions, which not available for scalar.
+Because of the SIMD computing, of course. But there's more to it: 
+heavy loops unrolling, specific instructions, which aren't available for **scalar WTF**.
```
Comment (homm) [inline README.md]: scalar data types

File: README.md [hunk 15]
```diff
@@ -83,89 +85,87 @@ Operation               | Filter  | IM   | Pillow| SIMD SSE4| SIMD AVX2| Skia 53
...
 
 ## Installation
 
-In general, you need to do `pip install pillow-simd` as always and if you
-are using SSE4-capable CPU everything should run smoothly.
-Do not forget to remove original Pillow package first.
-
-If you want the AVX2-enabled version, you need to pass the additional flag to C
-compiler. The easiest way to do that is define `CC` variable while compilation.
+If there's a copy of the original Pillow installed, it has to be removed first.
```
Comment (homm) [inline README.md]:
``` bash
$ pip uninstall -y pillow
```

File: README.md [hunk 16]
```diff
@@ -83,89 +85,87 @@ Operation               | Filter  | IM   | Pillow| SIMD SSE4| SIMD AVX2| Skia 53
...
+In general, you need to run `pip install pillow-simd`,
```
Comment (homm) [inline README.md]: In general, it's as simple as running `pip install pillow-simd`

File: README.md [hunk 17]
```diff
@@ -83,89 +85,87 @@ Operation               | Filter  | IM   | Pillow| SIMD SSE4| SIMD AVX2| Skia 53
...
-Do not forget to remove original Pillow package first.
-
-If you want the AVX2-enabled version, you need to pass the additional flag to C
-compiler. The easiest way to do that is define `CC` variable while compilation.
+If there's a copy of the original Pillow installed, it has to be removed first.
+In general, you need to run `pip install pillow-simd`, 
+and if you're using SSE4-capable CPU everything should run smoothly.
+If you'd like to install the AVX2-enabled version, 
+you need to pass the additional flag to a C compiler. 
+The easiest way to do so is to define the `CC` variable **while -> during?** the compilation.
```
Comment (homm) [inline README.md]: during — ok

**Review by homm — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (dayton1987) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (dayton1987) [reply README.md:23]: done

File: README.md (hunk 3 above)
Comment (dayton1987) [reply README.md]: corrected

File: README.md (hunk 4 above)
Comment (dayton1987) [reply README.md]: erased

File: README.md (hunk 5 above)
Comment (dayton1987) [reply README.md]: yup

File: README.md (hunk 6 above)
Comment (dayton1987) [reply README.md]: removed

File: README.md (hunk 7 above)
Comment (dayton1987) [reply README.md:59]: reconfigured the paragraph

File: README.md (hunk 9 above)
Comment (dayton1987) [reply README.md]: +

File: README.md (hunk 8 above)
Comment (dayton1987) [reply README.md]: erased

File: README.md (hunk 10 above)
Comment (dayton1987) [reply README.md]: reverted

File: README.md (hunk 11 above)
Comment (dayton1987) [reply README.md]: +

File: README.md (hunk 12 above)
Comment (dayton1987) [reply README.md]: transformed into a codeblock

File: README.md (hunk 13 above)
Comment (dayton1987) [reply README.md]: dumped

File: README.md (hunk 14 above)
Comment (dayton1987) [reply README.md]: fixed

File: README.md (hunk 16 above)
Comment (dayton1987) [reply README.md]: Shouldn't it be `$ pip install pillow-simd` instead?

File: README.md (hunk 15 above)
Comment (dayton1987) [reply README.md]: fxied

File: README.md (hunk 17 above)
Comment (dayton1987) [reply README.md]: done

**Review by dayton1987 — COMMENTED** [review]
Pillow-SIMD Review — Answered

**Review by homm — APPROVED** [review]

**Comment by homm** [comment]
Thanks!
Merged in https://github.com/uploadcare/pillow-simd/tree/simd/info

**Pull request closed without merging**